tools/devtool

```diff
@@ -170,6 +193,75 @@ ok_or_die() {
     [[ $code -eq 0 ]] || die -c $code "$@"
 }
 
+validate_shield() {
+    shield=$1
+    SYSFS_MOUNTPOINT=$(cat /proc/mounts | grep "^sysfs \/.\+ sysfs .*$" | cut -d' ' -f2)
+    CPUSET_MOUNTPOINT=$(cat /proc/mounts | grep -P "^[^ ]+ (/.+) (?:cpuset |cgroup (?:[^ ]*,)?cpuset[, ])" | cut -d' ' -f2 | head -n 1)
+
+    run_devctr \
+    --user "$(id -u):$(id -g)" \
+    --workdir "$CTR_FC_ROOT_DIR" \
+    --mount type=bind,source="$SYSFS_MOUNTPOINT",target="/host_sys" \
+    --mount type=bind,source="$CPUSET_MOUNTPOINT",target="/host_cpuset" \
+    -- \
+    python3 -c "
+import tools.shield_utils as su
+su.validate(\"$shield\")"
+    ok_or_die
+}
+
+shield_environment() {
+    numa_node_id=$1
+    cpus_count=$2
+    if [ ! -z $numa_node_id ] && [ ! -z $cpus_count ]; then
+        # Locate sysfs.
+        sysfs_mountpoint=$(cat /proc/mounts | grep "^sysfs \/.\+ sysfs .*$" | cut -d' ' -f2 | head -n 1)
+        cpuset_mountpoint=$(cat /proc/mounts | grep -P "^[^ ]+ (/.+) (?:cpuset |cgroup (?:[^ ]*,)?cpuset[, ])" | cut -d' ' -f2 | head -n 1)
+
+        # Get available cpuset cpus list.
+        CPUSET_CPUS=$(run_devctr \
+        --user "$(id -u):$(id -g)" \
+        --workdir "$CTR_FC_ROOT_DIR" \
+        --mount type=bind,source="$sysfs_mountpoint",target="/host_sys" \
+        --mount type=bind,source="$cpuset_mountpoint",target="/host_cpuset" \
+        -- \
+        python3 -c "import tools.shield_utils as su
```

Comment:
My feeling is that writing the shielding logic in python introduces some complicated interactions between bash and python. Not just here, but in multiple places. I understand that python is easier to use in some situations, but I think we  could simplify that logic a lot sunch that python wouldn't make a big difference, or maybe just use `cset shield`. In this case maybe we could consider rewriting the shielding logic in bash.

Reply:
We can not simply use `cset shield` because of few considerations that I thought are somehow blocking the progress. Running `cset shield` on host means bringing undesired host dependencies on python and `cset`.  Running `cset shield` within docker container is posible only if we patch the `cset shield` to actually look within custom directories, where the shared host sysfs and cgroup cpuset are, other than the mountpoints defined in `/proc/mounts` for the docker container counterpart of sysfs and cgroup cpuset. The implementation details can be later changed if we decide that we must simplify by other means and adjust to `cset shield`. Please let me know what are your thoughts on this.

Reply:
Also, using `cset shield` implies having an intrusive view over the topology of the platform, where shielded tests will run. This means that besides knowing how many NUMA nodes are available on the platform and how many CPUs every node has, we must find out which NUMA node are online, what is the NUMA node cpumask, and then choose a subset of the cpulist for a specific online NUMA node, depending on the requirements of the test. Again, for the NUMA node memory module similar steps must be followed. Most of these steps are automated by this PR. A descriptive error is returned if the steps are not satisfied.

Reply:
If we move the shielding logic outside docker, the extra dependency on `cset` would be hurting. If we decide to run the shielding logic in docker, `cset` might help. It would still bring some functionality that we could use. Not `cset shield` specifically, but `cset`.

As we discussed offline, first we should decide whether to have the shielding logic inside or outside docker.